### DISCUSSION TITLE
Failed to initiate the GitHub login process. Please try again.

### DISCUSSION THREAD

**Discussion by jenhaohowardkuo** [community/community#128192] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

Please Kindly help me resolve this problem for GitHub Copilot plugin.

I can use the plugin in my own laptop, but can't signin on the desktop in my company.

I've check all available discussion in YouTrack and stackoverflow, but no solution works for me.

I've also tried that Visual Studio is fine to suggest in-line and is fine to use copilot chat, so i think proxy is not the problem.

**Version** 
IntelliJ IDEA 2024.1.3 (Ultimate Edition)

Build #IU-241.17890.1, built on June 4, 2024

Runtime version: 17.0.11+1-b1207.24 amd64

VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.

Windows 10.0

com.github.copilot (1.5.6.5692)

**Copilot policies** 
Copilot Chat in the IDE: Enabled

**Steps to Reproduce** 
Start IntelliJ

Click : Main Menu (on the left top corner) -> Tools -> GitHub Copilot -> Login to GitHub

**Expected Result** 
Should get login modal

**Actual Result** 
No login modal

![snapshot fail to login](https://github.com/community/community/assets/146712202/fae33a53-0195-424f-ac10-6f0ee2fee1b5)

snapshot fail to login.PNG

**idea.log** 
2024-06-12 00:49:27,752 [ 6302] WARN - [#](https://youtrack.jetbrains.com/issue/IDEA-354789#steps-to-reproduce)com.github.copilot.chat.precondition.PreconditionsService - Error occurred trying to get preconditions

org.jetbrains.concurrency.MessageError: service is unavailable

**Comment by jenhaohowardkuo** [comment]
Copilot plugin update to latest version today: 1.5.7.5758
The issue still exists

**Reply by jenhaohowardkuo** [reply]
The idea.log for new version copilot plugin - 1.5.7.5758:
2024-06-13 09:08:23,538 [   3310]   WARN - #com.github.copilot.status.InspectionWidgetStatusProvider - Failed to check file status
org.jetbrains.concurrency.MessageError: service is unavailable

**Comment by jenhaohowardkuo** [comment]
Before 6/13, my personal laptop is OK to use plugin, and my PC in company is not work.
Start from 6/13 plugin update, both personal laptop and PC in company can not use the plugin now.

**Reply by lixiangyiIsNo1** [reply]
我尝试了2024.2的测试版webstorm, copilot也不能工作, 不知道到底是谁的问题

**Comment by hstaudacher** [comment]
Thanks for reporting this issue. I've opened an issue and we will have a look at this very shortly.

**Reply by lixiangyiIsNo1** [reply]
在最新的1.5.8.5775版本中, 我使用WebStorm 2024.1.4版本的IDE仍然不能工作, copilot的上下文还是不能生成, 没有任何报错提示

**Reply by Vishal008-Vichu** [reply]
@hstaudacher Is there any update on the fix or troubleshooting steps? I am facing the same error in IntelliJ IDE, tried all the workarounds in the comments. nothing works.
Ref: https://github.com/orgs/community/discussions/142354

**Comment by jenhaohowardkuo** [comment]
Still got same error after 6/14 update: Failed to initiate the GitHub login process. Please try again.
The idea.log for new version copilot plugin - 1.5.8.5775:
2024-06-14 09:15:33,604 [   4739]   WARN - #com.github.copilot.status.InspectionWidgetStatusProvider - Failed to check file status
org.jetbrains.concurrency.MessageError: service is unavailable

**Reply by jenhaohowardkuo** [reply]
Tried uninstall & re-install plugin again and still not work, and the error in idea.log is still the same.

**Comment by jenhaohowardkuo** [comment, marked as answer]
For the user whose plugin not works on windows,
notice that if ur IntelliJ is not 'run as administrator' but copilot agent at following path is 'run as administrator':
%APPDATA%\JetBrains\JetBrains\IntelliJIdea2024.1\plugins\github-copilot-intellij\copilot-agent\bin\copilot-agent-win.exe
The plugin's nodejs agent maybe does not have permission and will not be triggered, then plugin will not work.
Try to run IntelliJ and set copilot agent both with 'run as administrator' or both without the setting.

Copilot team, please try to avoid this problem.
No user modifies the agent as 'run as administrator' manually.
Maybe something ran by plugin change this setting?

**Reply by ashutoshsweng** [reply]
This solution doesn't work for me. Both IntelliJ and copilot agents are running without administrator setting.
Is there any other solution to this problem?

**Reply by RoundDev** [reply]
Same here. It didn't fix the problem for me.

**Reply by machariandirangu5982-create** [reply]
run pycharm /vs as administrator

**Comment by formbay-sanil** [comment]
I had to install Nodejs to make the co-pilot work.

**Reply by nusagates** [reply]
Same issue here. I am using latest version of Intellij IDE Ultimate Edition  & Github Copilot.
![Screenshot_7](https://github.com/user-attachments/assets/de168dac-fd8c-408c-98bc-ab7996a9a331)

**Reply by volceri** [reply]
Same here, installed node and it worked

**Comment by adielashrov** [comment]
Hey,
Running Pycharm as the administrator has solved the issue for me (for now).

Adiel

**Comment by romulomastelari** [comment]
This is happening to me now on 08/14, any solution to this problem? I just downloaded a copilot update and now it's giving this error.

**Reply by lancemonotone** [reply]
I am on PHPStorm and had this issue. I upgraded to the latest (2024.2) and I can now log in.

**Comment by stas-bob** [comment]
For me, the problem was that path ..\testProj\build\idea-sandbox\IU-2024.2.0.1\plugins\github-copilot-intellij\copilot-agent\native\win32-x64\copilot-language-server.exe was too long for windows. dont put your project in a deep folder structure or with long folder names

**Comment by phil-rice** [comment]
I have this issue as well. I don't have 'run as administrator' enabled.  The path was
C:\Users\USER\AppData\Roaming\JetBrains\IntelliJIdea2024.2\plugins\github-copilot-intellij\copilot-agent\native\win32-x64

![Screenshot 2024-10-19 075353](https://github.com/user-attachments/assets/0b3c4352-a49b-40c1-8020-bf154d2173fe)

**Reply by phil-rice** [reply]
In my case I fixed it by uninstalling the plugin. I then noticed that the folder in app data was still there. So reinstalling the plugin had no effect. For reference this folder for me is C:\Users\USER\AppData\Roaming\JetBrains\IntelliJIdea2024.2\plugins\github-copilot-intellij

It took some effort (rebooting the machine) to delete the folder as it was anchored by a running process somehow.

However once I had removed the folder and reinstalled the plugin, all worked fine

**Reply by YingwenDENG** [reply]
Tried this approach. But somehow it still doesn't work.

**Reply by JokerMas** [reply]
it doesnt work for me. now i try to delete everything and still doesnt work. i only have one last step is to reset my pc

**Comment by rick-seamless** [comment]
reinstalling the plugin worked for me

edit: just temporarily

**Comment by Leofierus** [comment]
![image](https://github.com/user-attachments/assets/64e729b3-43ac-4880-a52e-23d99039ebbe)
 Uninstalling and reinstalling the plugin works for me (pycharm) for a day or so but then I have to redo the process again everytime it stops working and throwing errors. HELP!

**Reply by jmediatechnology** [reply]
Windows guy with PHPStorm here, solved it by navigating to https://nodejs.org/en/download + installing a prebuilt nodejs. 
Uninstalling Github Copilot from Settings > Plugins. 
Rebooting. 
Installing the latest Github Copilot plug-in version from: 
https://plugins.jetbrains.com/plugin/17718-github-copilot/versions#tabs

**Comment by diomeh** [comment]
I'm running NixOS and facing into the same problem. Any way to go about providing an alternative login method? Say a login link and manually inputting a login code?

**Reply by zajca** [reply]
Did you have any luck? I do have same problem with nixos

**Reply by diomeh** [reply]
@zajca Nope, ultimately gave up on copilot and switched to JetBrains AI.

**Reply by zajca** [reply]
> @zajca Nope, ultimately gave up on copilot and switched to JetBrains AI.

I have actually find out, it's mentioned here, that plugin needs nodejs.

**Reply by Valkryst** [reply]
I ran into this issue as well, and I can confirm that [installing NodeJS](https://nixos.wiki/wiki/Node.js) and restarting the IDE resolved the issue.

**Reply by F-Joachim** [reply]
@zajca @Valkryst - Thanks a lot for your valuable posts.

I am running NixOS unstable (26.05) and I can confirm that the plugin works after installing nodejs 🥳

**Comment by Hardt-Coded** [comment]
Had the same issue on a windows machine. Find out, that the virus scanner was the problem. You can check this, by try to start the exe file in the plugin folder. On my end it was denied. Moved the plugins to another folder (https://www.jetbrains.com/help/rider/Directories_Used_by_the_IDE_to_Store_Settings_Caches_Plugins_and_Logs.html).

After moving to another folder, which was not highly watched by the virus scanner, it worked. Hope that helps.

**Comment by tigreci** [comment]
Same issue now in ubuntu 24.04
IntelliJ IDEA 2025.1.4.1 (Community Edition)
Build #IC-251.27812.49, built on July 24, 2025
Source revision: 735cc3bd3c92f
Runtime version: 21.0.7+9-b895.130 amd64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.awt.X11.XToolkit
Linux 6.14.0-27-generic
Ubuntu 24.04.2 LTS; glibc: 2.39
Kotlin plugin: K2 mode
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 2048M
Cores: 4
Registry:
  ide.experimental.ui=true
  llm.show.ai.promotion.window.on.start=false
Non-Bundled Plugins:
  Lombook Plugin (251.27812.12)
Kotlin: 251.27812.49-IJ
Current Desktop: ubuntu:GNOME

Plugin version: 1.5.52-243
<img width="308" height="642" alt="image" src="https://github.com/user-attachments/assets/d52c8774-8721-4109-8acf-695047962e4d" />

**Comment by kaosko** [comment]
Same issue on Fedora 41, worked fine before updating to IntelliJ IDEA 2025.1.4.1 (Community Edition) and the latest copilot update.

**Comment by tigreci** [comment]
It was fixed by accessing the GitHub profile and selecting some options
that weren’t previously checked. Apparently, new configuration parameters
were added to the user’s GitHub profile that hadn’t been set. After doing
that, it worked.

**Reply by tigreci** [reply]
It was fixed by accessing the GitHub profile and selecting some options that weren’t previously checked. Apparently, new configuration parameters were added to the user’s GitHub profile that hadn’t been set. After doing that, it worked.

**Reply by kaosko** [reply]
Yes, arrived at exactly the same solution. After visiting Copilot settings page on Github, then restarting IDE it started working for me.

**Reply by unraveller** [reply]
Sharing the exact setup would help others

**Reply by tigreci** [reply]
If you're experiencing login issues with the GitHub Copilot plugin, the problem might be due to new configuration fields added to the Copilot settings page. These fields are located here: https://github.com/settings/copilot/features.

If these fields are not properly set, authentication may fail and the plugin won't work as expected.

✅ Steps to resolve the issue:

Go to GitHub Copilot Settings.

Make sure the necessary options are enabled:

Copilot is activated for the environment you're using (e.g., Visual Studio Code).

Personalized suggestions are allowed if required by the plugin.

Accept any updated terms of service or privacy settings if prompted.

Save changes and restart your development environment.

💡 Extra tip: If the issue persists, try logging out of GitHub from your IDE, clearing any cached credentials, and logging back in after updating the settings.

Sharing this with your team can save them time and headaches. Let me know if you'd like help turning this into a README or internal guide!

**Comment by f1v3** [comment]
I'm experiencing the same problem and so far none of the fixes here worked.
We switched to using Virtual Machines at my company and since then, the problem started showing.

**Comment by Dowon-Kang** [comment]
Just update 
top right side, 
 setting -> plugin -> github copilot update

**Reply by f1v3** [reply]
Updated it several times since my post, but the problem is still showing.

**Comment by JokerMas** [comment]
<img width="1340" height="1164" alt="image" src="https://github.com/user-attachments/assets/1149178b-1e6d-4495-9b5d-57071dfeaac1" />
i keep getting this error when try to login into github copilot  in intelliJ : Failed to initiate the GitHub login process. Please try again.
here is the mode i have try : 
- i checked run as admin on intellij and copilot both run on no admin.
- I reinstall the plugin also didn't work
- i delete intellij and reinstall it didn't work(i also remove the folder jetbrains folder in local and roaming)
- i check my network nothing blocking and also copilot is working on vscode it just not working on intellij
- i have node js version 21 installed
- i try to connect to difference wifi it didn't work.
- i checked idea.log i see nothing related to copilot

can someone help me
